ksql-common/src/main/java/io/confluent/ksql/schema/ksql/types/SqlStruct.java

```diff
@@ -52,6 +53,12 @@ private SqlStruct(final List<Field> fields) {
     return fields;
   }
 
+  public Optional<Field> field(final String name) {
+    return fields.stream()
```

Comment:
what do you think about having `SqlStruct` create an `ImmutableMap`? If we're already making a copy of fields and iterating through the entire array then there shouldn't be a problem by just indexing it by name to make field fetching more efficient.

Reply:
done.